dev-server/ui-playground/components/patterns/SharedButtonPatterns.js

```diff
@@ -0,0 +1,133 @@
+            />
+          </PatternExample>
+        </PatternExamples>
+      </Pattern>
+
+      <Pattern title="LabeledButton">
+        <p>A button with a label (text) and optionally an icon.</p>
+        <PatternExamples title="Label only">
+          <PatternExample details="Basic usage">
+            <LabeledButton>Edit</LabeledButton>
+          </PatternExample>
+
+          <PatternExample details="Pressed">
+            <LabeledButton pressed>Edit</LabeledButton>
+          </PatternExample>
+
+          <PatternExample details="Expanded">
+            <LabeledButton expanded>Edit</LabeledButton>
+          </PatternExample>
+
+          <PatternExample details="Disabled">
+            <LabeledButton disabled>Edit</LabeledButton>
+          </PatternExample>
+
+          <PatternExample details="Primary variant">
+            <LabeledButton variant="primary">Edit</LabeledButton>
+          </PatternExample>
+        </PatternExamples>
+
+        <PatternExamples title="Label and icon">
+          <PatternExample details="Basic usage">
+            <LabeledButton icon="profile">Edit User</LabeledButton>
+          </PatternExample>
+
+          <PatternExample details="Pressed">
+            <LabeledButton icon="profile" pressed>
+              Edit User
+            </LabeledButton>
+          </PatternExample>
+
+          <PatternExample details="Expanded">
+            <LabeledButton icon="profile" expanded>
+              Edit User
+            </LabeledButton>
+          </PatternExample>
+
+          <PatternExample details="Disabled">
+            <LabeledButton icon="profile" disabled>
+              Edit User
+            </LabeledButton>
+          </PatternExample>
+
+          <PatternExample details="Right icon">
+            <LabeledButton icon="profile" iconPosition="right">
+              Edit User
+            </LabeledButton>
+          </PatternExample>
+
+          <PatternExample details="Primary variant">
+            <LabeledButton icon="profile" variant="primary">
+              Edit User
+            </LabeledButton>
+          </PatternExample>
+        </PatternExamples>
+      </Pattern>
+
+      <Pattern title="LinkButton">
+        <p>A button styled to look like a link (anchor tag).</p>
+        <PatternExamples>
+          <PatternExample details="Basic usage">
+            <LinkButton>Show replies (10)</LinkButton>
+          </PatternExample>
+
+          <PatternExample details="Pressed">
+            <LinkButton pressed>Show replies (10)</LinkButton>
+          </PatternExample>
+
+          <PatternExample details="Expanded">
+            <LinkButton expanded>Show replies (10)</LinkButton>
+          </PatternExample>
+
+          <PatternExample details="Disabled">
+            <LinkButton disabled>Show replies (10)</LinkButton>
+          </PatternExample>
+
+          <PatternExample details="Primary variant">
+            <LinkButton variant="primary">Show replies (10)</LinkButton>
```

Comment:
What's the use case for the `primary` variant of a `LinkButton`?

Reply:
It gets it for free via `ButtonBase`. But if it's distracting, I could do one or more of these things:

* Not render an example in the pattern library
* Explicitly disallow the `variant` prop for `LinkButton` (or otherwise restrict its values to not include `primary` in the future if other variants are needed)

Happy to do whatever here; we have lots of flexibility.